addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,241 @@
+                        )
+                    )
+                -- LEFT to allow keeping all aml even those without relation. The NULL will symbolize no relation
+                LEFT JOIN account_account_tag_account_tax_repartition_line_rel rep_tags
+                ON rep_tags.account_tax_repartition_line_id = tax_to_rep_line.id
+            ),
+
+            -- 1.b) Handle tax line: relation aml <-> tag, if no relation, tag is NULL
+            tax_aml_id_rep_tag_to_insert AS (
+                SELECT aml.id AS aml_id, rep_tags.account_account_tag_id AS tag_id
+                FROM account_move_line aml
+                JOIN account_tax_repartition_line rep_ln ON rep_ln.id = aml.tax_repartition_line_id
+                LEFT JOIN account_account_tag_account_tax_repartition_line_rel rep_tags
+                ON rep_tags.account_tax_repartition_line_id = aml.tax_repartition_line_id
+                WHERE aml.company_id = %(company_id)s AND aml.date >= %(date_from)s
+            ),
+
+            base_and_tax_aml_tag_id AS (
+                SELECT aml_id, tag_id FROM base_aml_id_rep_tag_to_insert
+                UNION
+                SELECT aml_id, tag_id FROM tax_aml_id_rep_tag_to_insert
+            ),
+
+            -- 2) Delete previous tags from amls
+            delete_statement AS (
+                DELETE FROM account_account_tag_account_move_line_rel aml_tags
+                USING base_and_tax_aml_tag_id btat
+                WHERE btat.aml_id = aml_tags.account_move_line_id
+                RETURNING account_move_line_id AS aml_id
+            ),
+
+            -- 3) Insert new tag on amls
+            insert_statement AS (
+                INSERT INTO account_account_tag_account_move_line_rel
+                SELECT btat.aml_id, btat.tag_id
+                FROM base_and_tax_aml_tag_id btat
+                WHERE btat.tag_id NOTNULL
+                RETURNING account_move_line_id AS aml_id
+            ),
+
+            -- 4) Return impacted amls
+            impacted_aml as (
+                SELECT aml_id
+                FROM delete_statement
+                UNION
+                SELECT aml_id
+                FROM insert_statement
+            )
+
+            SELECT ARRAY_AGG(impacted_aml.aml_id)
+              FROM impacted_aml
+        """, params={
+            'date_from': date_from,
+            'company_id': company_id,
+        })
+        self.env.invalidate_all()
+        return self.env.cr.fetchone()[0]
+
+    def update_amls_tax_tags(self):
+        parent_taxes = self.env['account.tax'].search([
+            ('company_id', '=', self.company_id.id),
+            ('children_tax_ids', '!=', False),
+        ])
+        children_taxes = []
+        for tax in parent_taxes:
+            children_taxes += tax.children_tax_ids.ids
+        if len(children_taxes) > len(parent_taxes.children_tax_ids.ids):
+            raise UserError(_('Update with children taxes that are child of multiple parents is not supported.'))
```

Comment:
I think we should allow people to bypass that through a setting in the wizard or something.

Reply:
Could be considered as a later improvement indeed, won't do that in a FW :wink: